config/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+        Self::try_from(provider).expect("failed to extract from provider")
+    }
+
+    /// Attempts to extract a `Config` from `provider`, returning the result.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use foundry_config::Config;
+    /// use figment::providers::{Toml, Format, Env};
+    ///
+    /// // Use foundry's default `Figment`, but allow values from `other.toml`
+    /// // to supersede its values.
+    /// let figment = Config::figment()
+    ///     .merge(Toml::file("other.toml").nested());
+    ///
+    /// let config = Config::try_from(figment);
+    /// ```
+    pub fn try_from<T: Provider>(provider: T) -> Result<Self, figment::Error> {
+        let figment = Figment::from(provider);
+        let mut config = figment.extract::<Self>()?;
+        config.profile = figment.profile().clone();
+        Ok(config)
+    }
+
+    /// Returns the default figment
+    ///
+    /// The default figment reads from the following sources, in ascending
+    /// priority order:
+    ///
+    ///   1. [`Config::default()`] (see [defaults](#defaults))
+    ///   2. `foundry.toml` _or_ filename in `FOUNDRY_CONFIG` environment variable
+    ///   3. `FOUNDRY_` prefixed environment variables
+    ///
+    /// The profile selected is the value set in the `FOUNDRY_PROFILE`
+    /// environment variable. If it is not set, it defaults to `default`.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use foundry_config::Config;
+    /// use serde::Deserialize;
+    ///
+    /// let my_config = Config::figment().extract::<Config>();
+    /// ```
+    pub fn figment() -> Figment {
+        Config::default().into()
+    }
+
+    /// Returns the default figment enhanced with additional context extracted from the provided
+    /// root, like remappings and directories.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use foundry_config::Config;
+    /// use serde::Deserialize;
+    ///
+    /// let my_config = Config::figment_with_root(".").extract::<Config>();
+    /// ```
+    pub fn figment_with_root(root: impl Into<PathBuf>) -> Figment {
+        Self::with_root(root).into()
+    }
+
+    /// Creates a new Config that adds additional context extracted from the provided root.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use foundry_config::Config;
+    /// let my_config = Config::with_root(".");
+    /// ```
+    pub fn with_root(root: impl Into<PathBuf>) -> Self {
+        // autodetect paths
+        let paths = ProjectPathsConfig::builder().build_with_root(root);
+
+        Config {
+            src: paths.sources.file_name().unwrap().into(),
+            out: paths.artifacts.file_name().unwrap().into(),
+            libs: paths.libraries.into_iter().map(|lib| lib.file_name().unwrap().into()).collect(),
+            remappings: paths.remappings,
+            ..Config::default()
+        }
+    }
+
+    /// Returns the default config but with hardhat paths
+    pub fn hardhat() -> Self {
```

Comment:
Should we have a `fn dapptools()` too which aliases to `fn default()`?

Reply:
that's what `Config::default` uses